Wrong editor time displayed in all time summary

**Describe the bug**
The editor time displayed on the main page of the summary (in my case for Android Studio) is oddly low. If you click on a project that mainly used this editor, the time is higher than on the main page. This is strange as it should contain all the data from the subprojects.

**System information**
* Wakapi version `2.10.4`
* Operating system
    * If Linux: which distro? `Ubuntu 22.04.3 LTS`
    * If Docker: which image and tag? `ghcr.io/muety/wakapi:2.10.4`
* Database `Postgres`

Global Editor chart (All Time):
![grafik](https://github.com/muety/wakapi/assets/52172848/47a6a300-b9ed-4c11-ba02-6cc165b1607a)

Editor Chart of a project (All Time)
![Bildschirmfoto vom 2024-01-30 17-51-12](https://github.com/muety/wakapi/assets/52172848/41a567e0-72e2-45cd-8742-c4dad3677678)

Realized that just now. I'm getting errors like:

```
[ERROR] failed to save summary (n1try, 2024-02-01 00:00:00 +0100 CET, 2024-02-02 00:00:00 +0100 CET) - unsupported data type: [0xc001b80120]: Table not set, please set it like: db.Model(&user) or db.Table("users")
```

Will wait for your fix. Thanks a lot again!

--- a/repositories/summary.go
+++ b/repositories/summary.go
@@ -37,7 +37,7 @@ func (r *SummaryRepository) GetAll() ([]*models.Summary, error) {
 func (r *SummaryRepository) Insert(summary *models.Summary) error {
 
 	if err := r.db.Transaction(func(tx *gorm.DB) error {
-		if err := r.db.Create(summary).Error; err != nil {
+		if err := tx.Create(summary).Error; err != nil {
 			return err
 		}
 
@@ -66,7 +66,7 @@ func (r *SummaryRepository) Insert(summary *models.Summary) error {
 			itemsToCreate = append(itemsToCreate, item)
 		}
 
-		if err := r.db.Create(itemsToCreate).Error; err != nil {
+		if err := tx.Create(itemsToCreate).Error; err != nil {
 			return err
 		}
 

--- a/repositories/summary.go
+++ b/repositories/summary.go
@@ -36,28 +36,42 @@ func (r *SummaryRepository) GetAll() ([]*models.Summary, error) {
 
 func (r *SummaryRepository) Insert(summary *models.Summary) error {
 
-	itemsToCreate := []interface{}{summary}
+	if err := r.db.Transaction(func(tx *gorm.DB) error {
+		if err := r.db.Create(summary).Error; err != nil {
+			return err
+		}
 
-	// required due to setting gorm:"-" in the model definition
-	// see https://github.com/muety/wakapi/issues/600#issuecomment-1921723789
-	// see https://github.com/muety/wakapi/pull/592#discussion_r1450478355
-	for _, item := range summary.Machines {
-		itemsToCreate = append(itemsToCreate, item)
-	}
+		itemsToCreate := []*models.SummaryItem{}
 
-	for _, item := range summary.Languages {
-		itemsToCreate = append(itemsToCreate, item)
-	}
+		// required due to setting gorm:"-" in the model definition
+		// see https://github.com/muety/wakapi/issues/600#issuecomment-1921723789
+		// see https://github.com/muety/wakapi/pull/592#discussion_r1450478355
+		for _, item := range summary.Machines {
+			item.SummaryID = summary.ID
+			itemsToCreate = append(itemsToCreate, item)
+		}
 
-	for _, item := range summary.OperatingSystems {
-		itemsToCreate = append(itemsToCreate, item)
-	}
+		for _, item := range summary.Languages {
+			item.SummaryID = summary.ID
+			itemsToCreate = append(itemsToCreate, item)
+		}
 
-	for _, item := range summary.Editors {
-		itemsToCreate = append(itemsToCreate, item)
-	}
+		for _, item := range summary.OperatingSystems {
+			item.SummaryID = summary.ID
+			itemsToCreate = append(itemsToCreate, item)
+		}
 
-	if err := r.db.Create(itemsToCreate).Error; err != nil {
+		for _, item := range summary.Editors {
+			item.SummaryID = summary.ID
+			itemsToCreate = append(itemsToCreate, item)
+		}
+
+		if err := r.db.Create(itemsToCreate).Error; err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
 		return err
 	}
 

